src/ResourceManager/Compute/Commands.Compute/Extension/AzureDiskEncryption/DisableAzureDiskEncryption.cs

```diff
@@ -125,6 +125,72 @@ private Hashtable GetExtensionProtectedSettings()
             return null;
         }
 
+        private bool IsDiskEncryptionSettingPresent(VirtualMachine vm)
```

Comment:
I would suggest to just check encryption settings in VM model instead of looping through disks etc. In the caller, you can use two pass encryption if VM model has encryption settings otherwise use single pass extension

Reply:
I don't have further comments. Once this is changed, consider approved from my side.

Reply:
I'll replace the helper method with one that checks VM model, logic flow:   use existing extension version if it exists, if it doesn't exist check if vm model encryption settings exist, if so use dual pass, in all other cases use single pass.